### PR TITLE
Note non-standard behavior of view transitions for forms set to post

### PR DESCRIPTION
**Pull request by knpwrs**
#### Description (required)

This PR adds a note about the non-standard behavior of Astro view transitions for forms with `method` set to `post`.

#### Related issues & labels (optional)

For Astro release: `4.0.7`. See astro [#9466](PR https://github.com/withastro/astro/pull/9466).

### CODE REVIEW

**Comment by OliverSpeir** [comment]
Good catch! I agree this non standard behavior should be documented

Anyone can see the behavior in the console logs when submitting the forms here:
https://stackblitz.com/edit/github-fluv9b-sagjcv?file=src%2Fpages%2Findex.astro

**Comment by martrapp** [comment]
@OliverSpeir Cool Page! 👍🏼 
> Anyone can see the behavior in the console logs when submitting the forms here:
https://stackblitz.com/edit/github-fluv9b-sagjcv?file=src%2Fpages%2Findex.astro

You might want to add
```
  <hr>
  <form method="POST" action="/form"  enctype="application/x-www-form-urlencoded">
    <button>Submit with Transitions and enctype</button>
  </form>
```

**Review by martrapp — APPROVED** [review]
correct in content, can't comment on wording ;-)

**Comment by martrapp** [comment]
> For future Astro release: 4.x.x. See astro [#9466](PR https://github.com/withastro/astro/pull/9466).

The "future" release is 4.0.7

**Comment by knpwrs** [comment]
I've updated the PR description to match.

File: src/content/docs/en/guides/view-transitions.mdx [hunk 1]
```diff
@@ -376,6 +376,14 @@ The `<ViewTransitions />` router will trigger in-page transitions from `<form>`
 
 By default, your page will transition after a form is submitted, even if you are using `event.preventDefault()` to prevent this behavior.
 
+Please note that if your method is set to `post`, the default behavior of web browsers is to submit your form data encoded as as `application/x-www-form-urlencoded`, but the default behavior of Astro is to submit your form data as `multipart/form-data`. If you want to match the default behavior of web browsers make sure to set the `enctype` attribute accordingly:
```
Comment (sarah11918) [inline src/content/docs/en/guides/view-transitions.mdx:379]:
```suggestion
By default, Astro submits your form data as `multipart/form-data` when your `method` is set to `POST`. If you want to match the default behavior of web browsers, use the `enctype` attribute to submit your data encoded as `application/x-www-form-urlencoded`:
```

File: src/content/docs/en/guides/view-transitions.mdx [hunk 2]
```diff
@@ -376,6 +376,14 @@ The `<ViewTransitions />` router will trigger in-page transitions from `<form>`
...
+
+```astro title="src/components/Form.astro"
+<form action="/contact" method="post" enctype="application/x-www-form-urlencoded">
```
Comment (sarah11918) [inline src/content/docs/en/guides/view-transitions.mdx:382]:
```suggestion
<form action="/contact" method="POST" enctype="application/x-www-form-urlencoded">
```

Comment (knpwrs) [reply src/content/docs/en/guides/view-transitions.mdx:382]: It's a stylistic thing. The docs currently have both. I'll leave it to you to decide :)

image

**Comment by sarah11918** [comment]
Thanks, I'm going off of this: 
image

We may have missed some in our v4 docs roundup!

**Comment by knpwrs** [comment]
@sarah11918 I think that's for exported API route handlers, not for form method verbs.

**Comment by sarah11918** [comment]
Gotcha. If it is a stylistic choice, it probably makes sense for us to converge with all caps, just for consistency and having one nice rule for contributors!

**Review by sarah11918 — APPROVED** [review]
This looks great now, thank you so much and welcome to Team Docs!  :partying_face:

**Pull request merged**